core/src/main/java/org/apache/iceberg/RemoveSnapshots.java

```diff
@@ -190,6 +269,11 @@ public void commit() {
         .onlyRetryOn(CommitFailedException.class)
         .run(item -> {
           TableMetadata updated = internalApply();
+          if (cleanExpiredFiles && updated.refs().size() > 1) {
+            throw new UnsupportedOperationException(
+                "Cannot remove snapshots and clean files when there is more than 1 ref or multiple lineages");
```

Comment:
This needs to mention that it is incremental. `"Cannot incrementally clean files for tables with more than 1 ref"`

Reply:
Updated the message to reflect that this is a limitation for specifically incremental file cleanup